airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -753,18 +753,22 @@ def filter_by_state(self, stream_state: Mapping[str, Any] = None, record: Mappin
             if isinstance(record.get(self.cursor_field), str)
             else pendulum.from_format(str(record.get(self.cursor_field)), self.cursor_field_datetime_format)
         )
-        start_date = (
-            pendulum.from_format(str(stream_state.get(self.cursor_field)), self.cursor_field_datetime_format)
-            if stream_state
-            else self._start_date
-        )
-        cursor_value = max(start_date, record_value)
+        # default state value before all futher checks
+        state_value = self._start_date
+        # we should check the presence of `stream_state` to overcome `availability strategy` check issues
+        if stream_state:
+            state_value = stream_state.get(self.cursor_field)
+            # sometimes the state is saved as `EMPTY STRING` explicitly
+            state_value = (
+                self._start_date if str(state_value) == "" else pendulum.from_format(str(state_value), self.cursor_field_datetime_format)
+            )
+        # compare the state with record values and get the max value between of two
+        cursor_value = max(state_value, record_value)
```

Comment:
let's add unit test for new functionality.

Reply:
There is no new functionality, but ok, I'll cover this.

Reply:
This is new code which handle empty state, before this PR we haven't this feature. So, this is new functionality and should be covered by unit tests. Thanks! 🙂